enumerables.rb

```diff
@@ -0,0 +1,145 @@
+module Enumerable
+  def my_each
+    count = 0
+    while count < length
+      yield(self[count])
+      count += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    count = 0
+    while count < length
+      yield(self[count], count)
+      count += 1
+    end
+    self
+  end
+
+  def my_select
+    result = []
+    my_each do |item|
+      result << item if yield(item)
+    end
+    result
+  end
+
+  def my_all?
+    result = true
+    my_each do |item|
+      result = false unless yield(item)
+    end
+    result
+  end
+
+  def my_any?
+    result = false
+    my_each do |item|
+      result = true if yield(item)
+    end
+    result
+  end
+
+  def my_none?
+    result = false
+    my_each do |item|
+      result = true unless yield(item)
+    end
+  end
+
+  def my_count
+    return length unless block_given?
+
+    result = 0
+    my_each do |item|
+      result = true unless yield(item)
+    end
+    result
+  end
```

Comment:
- [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method doesn't consider the possibility when a number is given as an argument `(some_enumerable.my_count(5)).` It should return the count of all elements that match the argument, yours is returning true.